src/coreclr/src/jit/gentree.cpp

```diff
@@ -16858,6 +16858,9 @@ CORINFO_CLASS_HANDLE Compiler::gtGetStructHandleIfPresent(GenTree* tree)
             case GT_OBJ:
                 structHnd = tree->AsObj()->GetLayout()->GetClassHandle();
                 break;
+            case GT_BLK:
```

Comment:
We use this path only in `optIsCSEcandidate`, but later `optIsCSEcandidate` doesn't know how to work with these opcodes, so it doesn't produce any diffs.

Reply:
Interesting, normally the layouts used by `BLK` do not have a class handle.

Reply:
I usually get such BLK nodes from OBJ nodes for structs that do not contain GC pointers.
I'll check if I see them in master.

Reply:
Ah, I've managed to confuse myself. Yes, `BLK`s can have class handles now, it was my plan to have handles only in `OBJ`s but it didn't happen.